### PR TITLE
Build a Neo4j-backed Chatbot with TypeScript

### PR DESCRIPTION
**Pull request by adam-cowley**

### CODE REVIEW

File: asciidoc/courses/llm-chatbot-typescript/modules/2-chains/lessons/2-answer-generation/lesson.adoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+----
+====
+
+== Using the Chain
+
+You will be able to initialize and run the chain in your application with the following code:
+
+[source,typescript]
+----
+include::{repository-raw}/{branch}/{lab-solution}[tag=usage]
```
Comment (martinohanlon) [inline asciidoc/courses/llm-chatbot-typescript/modules/2-chains/lessons/2-answer-generation/lesson.adoc:99]: This code is already included in the starter code - https://github.com/neo4j-graphacademy/llm-chatbot-typescript/blob/main/src/modules/agent/chains/answer-generation.chain.ts#L24

Should it be? Or should there just be a placeholder?

File: asciidoc/courses/llm-chatbot-typescript/modules/1-project-setup/lessons/2-setup/lesson.adoc [hunk 2]
```diff
@@ -54,17 +55,51 @@ git clone https://github.com/neo4j-graphacademy/llm-chatbot-typescript
...
+cd llm-chatbot-typescript
 npm install --include=dev
 
 
-=== Starting the App
+=== Setting Environment Variables
+
+To connect to Neo4j, you will need to set your Neo4j Sandbox credentials as environment variables.
+
+To set environment variables in a Next.js application, you can create a `.env.local` file.
```
Comment (martinohanlon) [inline asciidoc/courses/llm-chatbot-typescript/modules/1-project-setup/lessons/2-setup/lesson.adoc:72]: Perhaps simplify and just include a .env.local

File: asciidoc/courses/llm-chatbot-typescript/modules/2-chains/lessons/3-evaluating-responses/lesson.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+----
+include::{repository-raw}/{branch}/{lab-filename}[tag=eval,indent=0]
+----
+
+The test uses a concatenation of the output of the evaluation chain and the original response, so if the evaluation does not return a _yes_, both responses of the is displayed in the test output to help debug the issue.
+
+== Prompt Iteration
+
+As you develop your application and test different models, you may find that you need to modify the prompt iteratively until the model returns a consistent response.
+Running the tests in _watch_ mode will allow you to receive instant feedback.
```
Comment (martinohanlon) [inline asciidoc/courses/llm-chatbot-typescript/modules/2-chains/lessons/3-evaluating-responses/lesson.adoc:40]: Include instructions on how to run in watch mode?

File: asciidoc/courses/llm-chatbot-typescript/modules/2-chains/lessons/2-answer-generation/lesson.adoc (hunk 1 above)
Comment (adam-cowley) [reply asciidoc/courses/llm-chatbot-typescript/modules/2-chains/lessons/2-answer-generation/lesson.adoc:99]: It was originally in this file and not in the solution.  I must have copied it over to the chains.  I don't think it does any harm being in both places

File: asciidoc/courses/llm-chatbot-typescript/modules/1-project-setup/lessons/2-setup/lesson.adoc (hunk 2 above)
Comment (adam-cowley) [reply asciidoc/courses/llm-chatbot-typescript/modules/1-project-setup/lessons/2-setup/lesson.adoc:72]: Including a `.env.example` is an established pattern. I'd rather be explicit about setting the credentials correctly than causing confusion by including the wrong ones.

File: asciidoc/courses/llm-chatbot-typescript/includes/test.adoc [hunk 4]
```diff
@@ -0,0 +1,21 @@
+= Testing your changes
+
+If you have followed the instructions, you should be able to run the following unit test to verify the response using the `npm run test` command.
+
+.Running the Test
+[source,sh,subs="attributes+"]
+----
+npm run test {test-filename}
+----
+
```
[the remark this answers is not in the document]
Comment (adam-cowley) [reply asciidoc/courses/llm-chatbot-typescript/includes/test.adoc:10]: Good idea, I'll have a think about this one

File: asciidoc/courses/llm-chatbot-typescript/modules/3-conversation-history/lessons/1-theory/lesson.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
+= Using conversation History
+:type: quiz
```
Comment (martinohanlon) [inline asciidoc/courses/llm-chatbot-typescript/modules/3-conversation-history/lessons/1-theory/lesson.adoc:2]: should this be a lesson?

**Pull request merged**